module/user-extension/helper.js

```diff
@@ -538,169 +573,307 @@ module.exports = class UserExtensionHelper {
+                        roleMap[userExtensionData[0].roles[userExtensionRoleCounter].roleId.toString()].entities = userExtensionData[0].roles[userExtensionRoleCounter].entities;
+                    }
+                }
 
-            if( ! userExtensionData.length > 0 ) {
                 return resolve({
-                    message : constants.apiResponses.USER_EXTENSION_NOT_FOUND,
-                    result : {}
+                    message: constants.apiResponses.USER_EXTENSION_FETCHED,
+                    result:
+                        _.merge(_.omit(userExtensionData[0], ["roles", "entityDocuments", "roleDocuments"]),
+                            { roles: _.isEmpty(roleMap) ? [] : Object.values(roleMap) })
                 });
+
+            } catch (error) {
+                return reject(error);
             }
+        })
+    }
 
-            let roleMap = {};
-            let entityMapToRelatedEntities = {};
-
-            if( userExtensionData[0].entityDocuments && userExtensionData[0].entityDocuments.length >0 ) {
-                
-                let projection = [
-                    "metaInformation.externalId", 
-                    "metaInformation.name", 
-                    "metaInformation.addressLine1",
-                    "metaInformation.addressLine2",
-                    "metaInformation.administration",
-                    "metaInformation.city",
-                    "metaInformation.country",
-                    "entityTypeId",
-                    "entityType"
-                ];
+    /**
+     * Get profile with entity details
+     * @method
+     * @name getEntities
+     * @param {Object} filterQueryObject - filtered data.
+     * @param {String} entityType - entity type
+     * @param {String} pageSize - Size of page.
+     * @param {String} pageNo - Recent page no.
+     * @param {String} search - search text.
+     * @returns {Object} 
+     */
 
-                for ( 
-                    let pointerToUser = 0; 
-                    pointerToUser < userExtensionData[0].entityDocuments.length ;
-                    pointerToUser ++ 
-                ) {
-                    let currentEntities = 
-                    userExtensionData[0].entityDocuments[pointerToUser];
-
-                    const relatedEntities = 
-                    await entitiesHelper.relatedEntities(
-                        currentEntities._id, 
-                        currentEntities.entityTypeId, 
-                        currentEntities.entityType, 
-                        projection
-                    );
+    static getEntities(filterQueryObject, entityType,pageSize,pageNo,searchText) {
```

Comment:
@aman-tunerlabs - Please expect userId here, status etc filters should be setup here